### PR TITLE
Feat/8 handle multiple databases and users

### LINKED ISSUE #8: Handle multiple users
Right now we have a single `init_config` method ( which should probably be renamed) that asks from the user a username and password and stores the credentials locally (see README).

We have to extend this concept to multiple users when we want to support multiple GENESIS instances. Furthermore, it would be nice to somehow try to improve the usage, maybe by checking for this file automatically and otherwise on import ask the user by prompt or support env vars, what ever makes more sense or feels like a better UX

### PR DESCRIPTION
**Pull request by pmayd**
but open for comments

closes #8

### CODE REVIEW

**Comment by pmayd** [comment]
PR can be reviewed work is done

**Comment by pmayd** [comment]
Will fix code quality later

**Review by MarcoHuebner — COMMENTED** [review]
Mostly looked at the notebooks and left some minor comments. The overall logic is IMO easy to use 👍

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -24,31 +24,31 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.10", "3.11"]
         os: [ubuntu-22.04, macOS-latest, windows-latest]
```
[the remark this answers is not in the document]
Comment (pmayd) [reply .github/workflows/run-tests.yaml:28]: What do you mean? How does using ubuntu latest help in testing different Python versions?

File: nb/00_Setup.ipynb [hunk 2]
```diff
@@ -0,0 +1,118 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f4d35be-2e17-4e1a-8da4-0a3075f9911f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install python-dotenv"
```
[the remark this answers is not in the document]
Comment (pmayd) [reply nb/00_Setup.ipynb:10]: We can but it is more my personal style to use it it is by no means required...true the notebook uses it which is bad style already basically it should be optional and the developer can decide how to use and call setup credentials. But maybe it is not that bad to add it as dependency I just like to keep the dependencies to the bare minimum

File: nb/00_Setup.ipynb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ee7969b6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# only execute if you want to delete your config file for test purposes\n",
+    "config.delete_config()\n",
```
[the remark this answers is not in the document]
Comment (pmayd) [reply nb/00_Setup.ipynb:36]: It was until I used it and forgot to comment again XD

**Comment by pmayd** [comment]
@MarcoHuebner complete overhaul, I decided to not support custom config paths any longer as it is too much work and makes the code only more complex.

I fixes all tests, they create a new temporary config and restore the old user config, so tests should NO LONGER affect existing user configs!

File: .github/workflows/run-tests.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcoHuebner) [reply .github/workflows/run-tests.yaml:28]: I just meant `ubuntu-22.04` -> `ubuntu-latest`, since this is the way we do it for the other OS's as well

**Comment by MarcoHuebner** [comment]
> I fixes all tests, they create a new temporary config and restore the old user config, so tests should NO LONGER affect existing user configs!

Alright, whatever works for the user in the end 😄 I also addressed overwriting the `config_dir` in [7ce62ad](https://github.com/CorrelAid/pystatis/pull/4/commits/7ce62ad64d8cbb5b2e612b3f2986b2262bfc5cb7), but via mocking the settings directory

**Pull request merged**